packages/react/src/textarea/Textarea.tsx

```diff
@@ -0,0 +1,77 @@
+import clsx from "clsx";
+import { type ComponentPropsWithRef, type ReactNode, forwardRef } from "react";
+import TextareaAutosize from "react-textarea-autosize";
+
+import type { ExtendProps } from "../utils";
+
+import { Box } from "../box";
+import * as styles from "./Textarea.css";
+
+type TextareaProps = ExtendProps<
+  ComponentPropsWithRef<"textarea">,
+  ComponentPropsWithRef<typeof Box>,
+  {
+    autosize?: boolean;
+    bottomSection?: ReactNode;
+    defaultValue?: string;
+    isDisabled?: boolean;
+    isInvalid?: boolean;
+    isResizeable?: boolean;
```

Comment:
having both `autosize` and `isResizeable` now feels a bit confusing

can we combine these two into a single prop called `resize: 'auto' | 'vertical' | 'none'`

Reply:
addressed bhai

Reply:
Bhai, I think we should keep two props - `AutoResize` and `resize`. 

1. `AutoResize` is a boolean value which will provide the functionality of auto resizing the Component.
2. `resize` is the prop that lets the user expand functionality using the bottom right 'expand' icon.  <img width="27" alt="image" src="https://github.com/optimizely-axiom/optiaxiom/assets/162153227/06441a04-8c40-4b98-9217-fd75a0a9e9e0">
 
 
 I can use a single prop to render the same functionality but I think having two separate props make sense in this issue bhai.
 
We can make the following change to resize prop - `resize: 'vertical' | 'none'`.

Reply:
we're kind of taking over the `resize` prop and extending its functionality. since in the case of auto sizing you don't really want the user to be able to manually control the size - so they're mutually exclusive